### ISSUE TITLE
Multiple conditions in one line and/or nested conditions

### ISSUE THREAD

**Issue by kschzt** [jsoverson/preprocess#4]
Hi,

Here's an amended ifdef test that fails with multiple conditions on one line, and with nested conditions. HTH.

```javascript
  'ifdef': function(test) {
    test.expect(6);

    var input,expected,settings;

    input = "a<!-- @ifdef NONEXISTANT -->b<!-- @endif -->c";
    expected = "ac";
    test.equal(pp.preprocess(input, { }), expected, 'Fail case, should not be included');

    input = "a<!-- @ifdef NODE_ENV -->b<!-- @endif -->c";
    expected = "abc";
    test.equal(pp.preprocess(input, { NODE_ENV: 'dev'}), expected, 'Success case, should be included');

    input = "a/* @ifdef NONEXISTANT */b/* @endif */c";
    expected = "ac";
    test.equal(pp.preprocess(input, { },'js'), expected, 'Fail case, should not be included');

    input = "a/* @ifdef DEFA && DEFB */b/* @endif */c";
    expected = "abc";
    test.equal(pp.preprocess(input, { DEFA: true, DEFB: true },'js'), expected, 'Fail case, should not be included');

    input = "a\n// @ifdef DEFA \nb\n// @ifdef DEFB\nc\n// @endif\n// @endif\nd";
    expected = "abcd";
    test.equal(pp.preprocess(input, { DEFA: true, DEFB: true },'js'), expected, 'Fail case, should not be included');

    input = "a/* @ifdef NODE_ENV */b/* @endif */c";
    expected = "abc";
    test.equal(pp.preprocess(input, { NODE_ENV: 'dev'},'js'), expected, 'Success case, should be included');

    test.done();
  },
```

**Comment by drasive** [comment]
+1, this would be really useful

**Comment by mikey0000** [comment]
I've just written a patch which should do one line multiples for ifdef and ifndef

**Comment by BendingBender** [comment]
Nested directive support was implemented in #61.
